diff --git a/soh/CMakeLists.txt b/soh/CMakeLists.txt
index 5c239ae13..97947378f 100644
--- a/soh/CMakeLists.txt
+++ b/soh/CMakeLists.txt
@@ -672,6 +672,7 @@ endif()
 add_dependencies(${PROJECT_NAME}
     ZAPDUtils
     libultraship
+    GenerateSohOtr
 )
 if(NOT CMAKE_SYSTEM_NAME MATCHES "NintendoSwitch|CafeOS")
 add_dependencies(${PROJECT_NAME}
@@ -762,7 +763,7 @@ configure_file( ${CMAKE_CURRENT_SOURCE_DIR}/../scripts/linux/appimage/soh.sh.in
 endif()
 
 find_program(CURL NAMES curl DOC "Path to the curl program.  Used to download files.")
-execute_process(COMMAND ${CURL} -sSfL https://raw.githubusercontent.com/gabomdq/SDL_GameControllerDB/master/gamecontrollerdb.txt -o ${CMAKE_BINARY_DIR}/gamecontrollerdb.txt OUTPUT_VARIABLE RESULT)
+execute_process(COMMAND cp ${CMAKE_SOURCE_DIR}/gamecontrollerdb.txt ${CMAKE_BINARY_DIR}/gamecontrollerdb.txt)
 
 if("${CMAKE_SYSTEM_NAME}" STREQUAL "Darwin")
 configure_file( ${CMAKE_CURRENT_SOURCE_DIR}/macosx/Info.plist.in ${CMAKE_BINARY_DIR}/macosx/Info.plist @ONLY)
